File: 1507-go-vue----/src/client/api_client.py
import os
from typing import List, Optional, Dict, Any

import requests


class APIClient:
    def __init__(self, base_url: Optional[str] = None):
        self.base_url = base_url or os.environ.get(
            "SERVER_URL", "http://localhost:8000"
        )

    def _request(self, method: str, path: str, **kwargs) -> Dict[str, Any]:
        url = f"{self.base_url}{path}"
        try:
            response = requests.request(method, url, timeout=10, **kwargs)
            response.raise_for_status()
            if response.content:
                return response.json()
            return {}
        except requests.exceptions.RequestException as e:
            raise RuntimeError(f"请求失败: {e}")

    def create_rider(self, rider_id: str) -> Dict[str, Any]:
        return self._request("POST", "/riders", json={"rider_id": rider_id})

    def list_riders(self, status: Optional[str] = None) -> List[Dict[str, Any]]:
        params = {"status": status} if status else {}
        return self._request("GET", "/riders", params=params)

    def get_rider(self, rider_id: str) -> Dict[str, Any]:
        return self._request("GET", f"/riders/{rider_id}")

    def heartbeat(self, rider_id: str) -> Dict[str, Any]:
        return self._request("POST", f"/riders/{rider_id}/heartbeat")

    def set_rider_status(self, rider_id: str, status: str) -> Dict[str, Any]:
        return self._request(
            "POST",
            f"/riders/{rider_id}/status",
            json={"status": status}
        )

    def accept_order(self, rider_id: str, order_id: str) -> Dict[str, Any]:
        return self._request(
            "POST",
            f"/riders/{rider_id}/accept-order",
            json={"order_id": order_id}
        )

    def create_order(self) -> Dict[str, Any]:
        return self._request("POST", "/orders")

    def list_orders(self, status: Optional[str] = None) -> List[Dict[str, Any]]:
        params = {"status": status} if status else {}
        return self._request("GET", "/orders", params=params)

    def get_order(self, order_id: str) -> Dict[str, Any]:
        return self._request("GET", f"/orders/{order_id}")

    def complete_order(self, order_id: str) -> Dict[str, Any]:
        return self._request("POST", f"/orders/{order_id}/complete")

    def reassign_order(self, order_id: str, rider_id: str) -> Dict[str, Any]:
        return self._request(
            "POST",
            f"/orders/{order_id}/reassign",
            json={"rider_id": rider_id}
        )

    def get_metrics(self) -> Dict[str, Any]:
        return self._request("GET", "/metrics")

File: 1472-go-vue----/client/api_client.go
package main

import (
	"bytes"
	"encoding/json"
	"fmt"
	"io/ioutil"
	"net/http"
	"taxisystem/common"
)

type APIClient struct {
	baseURL string
}

func NewAPIClient(baseURL string) *APIClient {
	return &APIClient{baseURL: baseURL}
}

func (c *APIClient) doRequest(method, path string, body interface{}) (*common.APIResponse, error) {
	var reqBody []byte
	var err error

	if body != nil {
		reqBody, err = json.Marshal(body)
		if err != nil {
			return nil, fmt.Errorf("序列化请求体失败: %w", err)
		}
	}

	url := c.baseURL + path
	req, err := http.NewRequest(method, url, bytes.NewBuffer(reqBody))
	if err != nil {
		return nil, fmt.Errorf("创建请求失败: %w", err)
	}

	req.Header.Set("Content-Type", "application/json")

	client := &http.Client{}
	resp, err := client.Do(req)
	if err != nil {
		return nil, fmt.Errorf("发送请求失败: %w", err)
	}
	defer resp.Body.Close()

	respBody, err := ioutil.ReadAll(resp.Body)
	if err != nil {
		return nil, fmt.Errorf("读取响应失败: %w", err)
	}

	var apiResp common.APIResponse
	if err := json.Unmarshal(respBody, &apiResp); err != nil {
		return nil, fmt.Errorf("解析响应失败: %w, 响应内容: %s", err, string(respBody))
	}

	return &apiResp, nil
}

func (c *APIClient) CreateVehicle(plate, driverName, driverPhone string, lat, lng float64) (*common.APIResponse, error) {
	req := common.CreateVehicleRequest{
		PlateNumber: plate,
		DriverName:  driverName,
		DriverPhone: driverPhone,
		Latitude:    lat,
		Longitude:   lng,
	}
	return c.doRequest("POST", "/api/v1/vehicles", req)
}

func (c *APIClient) ListVehicles() (*common.APIResponse, error) {
	return c.doRequest("GET", "/api/v1/vehicles", nil)
}

func (c *APIClient) GetVehicle(plate string) (*common.APIResponse, error) {
	return c.doRequest("GET", "/api/v1/vehicles/"+plate, nil)
}

func (c *APIClient) UpdateVehicleStatus(plate string, status common.VehicleStatus) (*common.APIResponse, error) {
	req := common.UpdateVehicleStatusRequest{Status: status}
	return c.doRequest("PUT", "/api/v1/vehicles/"+plate+"/status", req)
}

func (c *APIClient) UpdateVehicleLocation(plate string, lat, lng float64) (*common.APIResponse, error) {
	req := common.UpdateVehicleLocationRequest{
		Latitude:  lat,
		Longitude: lng,
	}
	return c.doRequest("PUT", "/api/v1/vehicles/"+plate+"/location", req)
}

func (c *APIClient) EstimateFare(pickup, dest common.Location) (*common.APIResponse, error) {
	req := common.EstimateFareRequest{
		PickupLocation: pickup,
		DestLocation:   dest,
	}
	return c.doRequest("POST", "/api/v1/orders/estimate", req)
}

func (c *APIClient) CreateOrder(passengerID, passengerPhone string, pickup, dest common.Location) (*common.APIResponse, error) {
	req := common.CreateOrderRequest{
		PassengerID:    passengerID,
		PassengerPhone: passengerPhone,
		PickupLocation: pickup,
		DestLocation:   dest,
	}
	return c.doRequest("POST", "/api/v1/orders", req)
}

func (c *APIClient) ListOrders() (*common.APIResponse, error) {
	return c.doRequest("GET", "/api/v1/orders", nil)
}

func (c *APIClient) GetOrder(orderID string) (*common.APIResponse, error) {
	return c.doRequest("GET", "/api/v1/orders/"+orderID, nil)
}

func (c *APIClient) AcceptOrder(orderID, plateNumber string) (*common.APIResponse, error) {
	req := common.AcceptOrderRequest{PlateNumber: plateNumber}
	return c.doRequest("POST", "/api/v1/orders/"+orderID+"/accept", req)
}

func (c *APIClient) StartTrip(orderID string, actualDistance float64, actualDuration int64, lowSpeedMinutes int) (*common.APIResponse, error) {
	req := common.StartTripRequest{
		ActualDistance:   actualDistance,
		ActualDuration:   actualDuration,
		LowSpeedMinutes:  lowSpeedMinutes,
	}
	return c.doRequest("POST", "/api/v1/orders/"+orderID+"/start", req)
}

func (c *APIClient) CompleteTrip(orderID string) (*common.APIResponse, error) {
	return c.doRequest("POST", "/api/v1/orders/"+orderID+"/complete", nil)
}

func (c *APIClient) SubmitRating(orderID string, stars int, content string) (*common.APIResponse, error) {
	req := common.SubmitRatingRequest{
		Stars:   stars,
		Content: content,
	}
	return c.doRequest("POST", "/api/v1/orders/"+orderID+"/rating", req)
}

func (c *APIClient) ListComplaints() (*common.APIResponse, error) {
	return c.doRequest("GET", "/api/v1/complaints", nil)
}

func (c *APIClient) HandleComplaint(complaintID string) (*common.APIResponse, error) {
	return c.doRequest("POST", "/api/v1/complaints/"+complaintID+"/handle", nil)
}

func (c *APIClient) GetMetrics() (*common.APIResponse, error) {
	return c.doRequest("GET", "/api/v1/metrics", nil)
}
